### ISSUE TITLE
How to generate tsne?

### ISSUE THREAD

**Issue by Blosslzy** [SCLBD/DeepfakeBench#72]
detector_name_list = [
    '/mntcephfs/lab_data/zhiyuanyan/benchmark_results/tsne/tsne_dict_xception_0_270.pkl',
    '/mntcephfs/lab_data/zhiyuanyan/benchmark_results/tsne/tsne_dict_xception_0.pkl',
]
 How to modify the path to generate tsne? Thank you very much!

**Comment by Blosslzy** [comment]
I reproduced UCF on your provided FF++, which is different from the results in the paper, but the retrained xception is slightly higher than the reported results. Is there any solution? Is it related to the training data? Thank you very much!
|情况|Method|CelebDFv2|DFDC|DFDC-P|
|官方|UCF|75.27|71.91|75.94|
|复现|UCF|73.05|69.46|72.00|
|官方|Xception|73.65|70.77|73.74|
|复现|Xception|75.45|71.17|72.78|

**Comment by Blosslzy** [comment]
I have noticed a weird operation on train.py 
```
def main():
    # parse options and load config
    with open(args.detector_path, 'r') as f:
        config = yaml.safe_load(f)
    with open('./training/config/train_config.yaml', 'r') as f:
        config2 = yaml.safe_load(f)
    config.update(config2)
```
I think this operation will replace the 1, 2, 3, 4 labels defined in ucf.yaml of FF-F2F, FF-DF, FF-FS to 1, 1, 1, 1 defined in train_config.yaml. Thus, the specific feature will become invalid. I replaced this operation with:
```
with open(args.detector_path, 'r') as f:
        config2 = yaml.safe_load(f)
    with open('/home/zbp/lzy/DeepfakeBench-main/training/config/train_config.yaml', 'r') as f:
        config = yaml.safe_load(f)
    config.update(config2)
```
The AUC results dropped from 75 to 73 when generalizing from FF++ to CelebDF-v2.

**Comment by Blosslzy** [comment]
Yes, I solved it by saving the prediction results and ground truth labels in a .pkl file.

**Comment by Ablott123** [comment]
Hello, can you give the relevant code?

**Comment by Ablott123** [comment]
> Yes, I solved it by saving the prediction results and ground truth labels in a .pkl file.
The pkl file I created cannot be used. There may be some problems with my code. Can I ask how to modify the code to save this file?

**Comment by BA-dp** [comment]
> 是的，我通过将预测结果和基本事实标签保存在 .pkl 文件中来解决它。

Hello, if it's convenient for you, could you please send a copy of the code to my email at maetetet@163.com? I would be extremely grateful.
